components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
 		blackListWhiteListService = new BlackListWhiteListService();
 	}
 
-	/**
-	 * GET method for creating an instance of QueriesResource
-	 *
-	 * @param address,requestedAccuracy
-	 *            representation for the new resource
-	 * @return an HTTP response with content of the created resource
-	 * @throws Exception
-	 */
-	@Deprecated
-	@POST
-	@Path("/Blacklist/{MSISDN}")
-	@Consumes("application/json")
-	@Produces("application/json")
-	public Response location(@PathParam("MSISDN") String msisdn, String jsonBody) throws Exception {
-		LOG.debug("location Triggerd  jsonBody :" + jsonBody + " jsonBody: " + jsonBody);
-		Gson gson = new GsonBuilder().serializeNulls().create();
-
-		final StringBuilder errorMSG = new StringBuilder();
-		final StringBuilder jsonreturn = new StringBuilder();
+	@GET
+	@Path("/search/{app}/{api}")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response search(@PathParam("app") String app, @PathParam("api") String api,
+						   @QueryParam("msisdn") String msisdn, @QueryParam("action") String action, @QueryParam("sp") String sp) {
 
-		errorMSG.append("{").append("\"Failed\":".intern()).append("{".intern());
-		errorMSG.append("\"messageId\":\"".intern()).append("Blacklist Number".intern()).append("\",".intern());
-		errorMSG.append("\"text\":\"".intern()).append("Blacklist Number could not be added to the system ".intern())
-				.append("\",");
-		errorMSG.append("\"variables\":\"".intern()).append(msisdn).append("\"");
-		errorMSG.append("}}");
+		Gson gson = new Gson();
```

Comment:
validate the incoming parameters before processing the business logic. Validate the mobile number by creating an annotation is the better way I guess
https://www.baeldung.com/spring-mvc-custom-validator